tests/integration/loki_tester/src/charm.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""A Charm to functionally test the Loki Operator."""
+import contextlib
+import json
+import logging
+from pathlib import Path
+from typing import List
+
+from charms.loki_k8s.v0.loki_push_api import LogProxyConsumer
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+from ops.pebble import ChangeError, Layer
+
+logger = logging.getLogger(__name__)
+LOGFILE = "/loki_tester_msgs.txt"
+
+
+@contextlib.contextmanager
+def _catch_quick_exit_errors():
```

Comment:
Ha cool! didn't know that existed. But it does not seem to allow you to check the args of the exception. Unlike a `pytest.raises`. That's a deal breaker here since I'm afraid other legitimate ChangeErrors might be raised.